fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -40,15 +72,167 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
+                                    translationsOptionsHeightDp = with(density) {
+                                        coordinates.size.height.toDp()
+                                    }
+                                },
+                            ) {
+                                ShowTranslationOptions(
+                                    onDismiss = {
+                                        translationsVisibility = true
+                                    },
+                                )
+                            }
+                        }
+                    }
+                }
             }
         }
     }
+
+    @Composable
+    private fun TranslationsAnimation(
+        translationsVisibility: Boolean,
+        density: Density,
+        translationsOptionsHeightDp: Dp,
+        content: @Composable AnimatedVisibilityScope.() -> Unit,
+    ) {
+        AnimatedVisibility(
+            visible = translationsVisibility,
+            enter = expandIn(
+                initialSize = {
+                    with(density) {
+                        IntSize(
+                            0,
+                            translationsOptionsHeightDp.roundToPx(),
+                        )
+                    }
+                },
+            ) + fadeIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ) + slideInHorizontally(),
+        ) {
+            content()
+        }
+    }
+
+    @Composable
+    private fun TranslationsOptionsAnimation(
+        translationsVisibility: Boolean,
+        density: Density,
+        translationsHeightDp: Dp,
+        translationsWidthDp: Dp,
+        content: @Composable AnimatedVisibilityScope.() -> Unit,
+    ) {
+        AnimatedVisibility(
+            visible = !translationsVisibility,
+            enter = expandIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+                initialSize = {
+                    with(density) {
+                        IntSize(
+                            0,
+                            translationsHeightDp.roundToPx(),
+                        )
+                    }
+                },
+            ) + fadeIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ) + slideInHorizontally(
+                initialOffsetX = { with(density) { translationsWidthDp.roundToPx() } },
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ),
+        ) {
+            content()
+        }
+    }
+
+    @Composable
+    private fun ShowTranslations(onDismiss: () -> Unit) {
+        TranslationsDialogBottomSheet(
+            onSettingClicked = onDismiss,
+        )
+    }
+
+    @Composable
+    private fun ShowTranslationOptions(onDismiss: () -> Unit) {
```

Comment:
Let's inline these instead so we don't even just continually passing callback parameters as their functionality grows.

Reply:
I would still prefer to see this change. At the very least, I find the naming of the Composable functions here should change to align with how we should be naming Composables: `TranslationsDialog`, `TranslationsOptionsDialog`, and I would like to see `onDismiss` be renamed to align with the expected param naming `onSettingClicked`, `onBackClicked`.